core/src/test/java/org/pentaho/di/core/row/value/ValueMetaBaseTest.java

```diff
@@ -1085,4 +1092,62 @@ public void testHashCodes() throws Exception {
     valueMetaString.type = ValueMetaInterface.TYPE_NONE;
     assertEquals( valueMetaString.hashCode( "any" ),  0 );
   }
+
+  @Test
```

Comment:
Please update these tests so that you call `setPreparedStatementValue()` with `ValueMetaInterface.TYPE_STRING` and verify that you get the expected values for the different use cases.